Update 3D Tiles Next readme (#546)
CesiumGS/3d-tiles: next/README.md

<p align="center"><img src="./figures/next_logo_dark.png" alt="3D Tiles Next" width="500px" align="center"></p>

# 3D Tiles Next

## Overview

> **3D Tiles Next** is a set of new capabilities for the future of 3D Tiles. Provided as extensions to the [3D Tiles 1.0](../specification) specification, these features may be incorporated into 3D Tiles 2.0 in the future.

Reviewer: ```suggestion
**3D Tiles Next** is a set of new capabilities for the future of 3D Tiles. Provided as draft extensions to the [3D Tiles 1.0](../specification) specification, these features may be incorporated into 3D Tiles 2.0 in the future.
```

|   | Extension |
|---|-----------|
| <img src="figures/content-gltf.png" width="600px" alt="Tileset with glTF content"> | [`3DTILES_content_gltf`](../extensions/3DTILES_content_gltf) (3D Tiles 1.0 extension) <br> Allows references to glTF models (`.gltf`, `.glb`) directly in `tile.content`, without intermediate formats. Provides easier integration with a wider ecosystem of software producing glTF assets. |
| <img src="figures/multiple-contents.png" width="600px" alt="Tileset with multiple contents"> | [`3DTILES_multiple_contents`](../extensions/3DTILES_multiple_contents) (3D Tiles 1.0 extension) <br> Allows storing more than one content model per tile, and in effect, per single volume of space. Contents can be organized in various ways — e.g. as map layers or arbitrary groupings — which becomes particularly useful when combined with content group metadata defined by `3DTILES_metadata`. |

> glTF content is the future of 3D Tiles, and may be used in place of glTF Tiles 1.0 internal formats (`.b3dm`, `.i3dm`, `.pnts`) in the future. Those internal formats remain available, but glTF offers forward-compatibility with a wider ecosystem of 3D content tools, and with any current and future extensions developed for the glTF format. For technical details on switching to glTF tile content, see the developer [migration guide](../extensions/3DTILES_content_gltf/MIGRATION_GUIDE.md) provided with `3DTILES_content_gltf`.

Reviewer 2: ```suggestion
glTF content is the future of 3D Tiles, and may be used in place of 3D Tiles 1.0 internal formats (`.b3dm`, `.i3dm`, `.pnts`) in the future. Those internal formats remain available, but glTF offers forward-compatibility with a wider ecosystem of 3D content tools, and with any current and future extensions developed for the glTF format. For technical details on switching to glTF tile content, see the developer [migration guide](../extensions/3DTILES_content_gltf/MIGRATION_GUIDE.md) provided with `3DTILES_content_gltf`.
```

| <img src="figures/feature-metadata.png" alt="Metadata in glTF" width="600"> | [`EXT_mesh_features`](https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features) (glTF 2.0 extension) <br> Defines 3D Metadata in glTF 2.0 assets. Combined with `3DTILES_content_gltf`, metadata is associated with subcomponents ("features") within a tile's content at various levels of granularity: <br><br>• *Per vertex:* Vertices may be grouped into features each having associated properties, similar to Batch Tables in 3D Tiles 1.0. <br>• *Per texel*: Textures storing property values and feature IDs allow detailed metadata even for optimized, low-poly geometry. <br>• *Per GPU instance:* GPU instances (defined by glTF extension `EXT_mesh_gpu_instancing`) may be assigned properties, similar to `.i3dm` files. |

> The 3D Metadata Specification also provides a [Semantic Reference](../specification/Metadata/Semantics), used to describe the semantic meanings of properties. Semantics may be general-purpose (e.g. ID, name, or description), related 3D Tiles (e.g. bounding volumes and occlusion data), or extended and customized for other domains.

Reviewer 2: ```suggestion
The 3D Metadata Specification also provides a [Semantic Reference](../specification/Metadata/Semantics), used to describe the semantic meanings of properties. Semantics may be general-purpose (e.g. ID, name, or description), related to 3D Tiles (e.g. bounding volumes and occlusion data), or extended and customized for other domains.
```

> <p align="center"><img src="./figures/next_logo.png" alt="3D Tiles Next" width="500px" align="center"></p>

Reviewer: Just a heads-up, this logo might be tweaked after some feedback with Patrick.

Author: Sounds good. If there's a canonical version elsewhere in the repository I can just link to that, or maybe this is the right place for it?

- [Tile Content](#tile-content): glTF 2.0 assets may be used directly as tile content, without intermediate formats, improving interoperability with 3D content and tooling ecosystems. Tiles may reference multiple contents — for organization, styling, and filtering — and contents may be collected into groups similar to map layers in mapping applications.
- [Implicit Tiling](#implicit-tiling): Common subdivision schemes and spatial index patterns may be declared without listing bounding volumes exhaustively. Reduces tileset size, and enables new optimizations including faster traversal, raycasting, random access, and spatial queries.
- [Metadata](#metadata): Metadata in 3D Tiles gains more expressiveness and flexibility, with a well-defined type system, new encoding options (e.g. JSON or binary), and a range of granularity options. Metadata may be associated with high-level objects like tilesets, tiles, or tile content groups, or with individual vertices and texels on glTF 2.0 geometry.

## Tile Content

> _**Overview:** glTF 2.0 assets may be used directly as tile content, without intermediate formats, improving interoperability with 3D content and tooling ecosystems. Tiles may reference multiple contents — for organization, styling, and filtering — and contents may be collected into groups similar to map layers in mapping applications._

Author: Ok – if this level of detail is otherwise good (i.e. we don't expect it to _become_ a particularly long document) then let's just remove the three bullets at the top. Note these are also repeated in the main README.md, but that seems more justifiable.

|   | Extension |
|---|-----------|
| <img src="figures/content-gltf.png" width="600px" alt="Tileset with glTF content"> | [`3DTILES_content_gltf`](../extensions/3DTILES_content_gltf) (3D Tiles 1.0 extension) <br> Allows references to glTF models (`.gltf`, `.glb`) directly in `tile.content`, without intermediate formats. Provides easier integration with a wider ecosystem of software producing glTF assets. |
| <img src="figures/multiple-contents.png" width="600px" alt="Tileset with multiple contents"> | [`3DTILES_multiple_contents`](../extensions/3DTILES_multiple_contents) (3D Tiles 1.0 extension) <br> Allows storing more than one content model per tile, and in effect, per single region of space. Contents can be organized in various ways — e.g. as map layers or arbitrary groupings — which becomes particularly useful when combined with content group metadata defined by `3DTILES_metadata`. |

> glTF content is the future of 3D Tiles, and may be used in place of glTF Tiles 1.0 internal formats (`.b3dm`, `.i3dm`, `.pnts`) in the future. Those internal formats remain available, but glTF offers forward-compatibility with a wider ecosystem of 3D content tools, and with any current and future extensions developed for the glTF format.

Author: Done ✅

In addition to size and performance benefits, these extensions provide improved interoperability with geospatial formats and libraries such as CDB, TMS, WMTS, and S2.

## Metadata

_**Overview:** Metadata in 3D Tiles gains more expressiveness and flexibility, with a well-defined type system, new encoding options (e.g. JSON or binary), and a range of granularity options. Metadata may be associated with high-level objects like tilesets, tiles, or tile content groups, or with individual vertices and texels on glTF 2.0 geometry._

The foundation of these new metadata extensions is the format-agnostic [3D Metadata Specification](../specification/Metadata). The specification defines key concepts including schemas, property types, storage formats, and semantic meanings of properties. Extensions for metadata in 3D Tiles and glTF 2.0 are available now, and other formats can be extended to support the common format of the 3D Metadata Specification in the future.

|   | Extension |
|---|-----------|

> | <img src="figures/metadata-granularity.png" alt="Metadata in 3D Tiles" width="600"> | [`3DTILES_metadata`](../extensions/3DTILES_metadata) (3D Tiles 1.0 extension) <br> Defines 3D Metadata for tilesets, tiles, and tile content groups. Complements other 3D Tiles Next extensions with additional metadata-related features: <br><br>• In `3DTILES_implicit_tiling` tile metadata may be stored compactly in binary subtree files, allowing efficient streaming of large global tilesets and their metadata. <br>• In `3DTILES_multiple_contents` each tile content may be associated with a group and with group metadata, enabling improved content organization, styling, and filtering. |

Reviewer: Just to check, is the extended metadata diagram needed? if not, would be good to remove it from the PR

Author: I'd included it becomes I think it's a helpful overview of how metadata works, and which extension should be used depending on the level of granularity required – could you say more about why it would be good to remove it?

If that's something about the size being too much in this context, we could scale it down:

![Screen Shot 2021-11-09 at 8 55 33 AM](https://user-images.githubusercontent.com/1848368/140970186-86ddf792-b8ee-40e0-8b4b-1116e261e923.png)

Reviewer: Oh sorry I somehow missed seeing it referenced in the diff, ignore me

For further use cases of feature metadata, see the [examples](https://github.com/CesiumGS/glTF/blob/proposal-EXT_mesh_features/extensions/2.0/Vendor/EXT_mesh_features/README.md#examples) in the `EXT_mesh_features` specification. Together, `3DTILES_metadata` and `EXT_mesh_features` allow flexible definitions of metadata at any level of granularity within a tileset. The illustration below provides an overview of these levels, in which the first levels ("Tileset", "Tiles", "Tile Content Groups") are provided by `3DTILES_metadata` and the lowest level ("Features") is provided by `EXT_mesh_features` in a glTF asset.

![Available metadata granularity, including tilesets, tiles, tile content groups, and features.](./figures/metadata-granularity-extended.png)

## Contributing

> 3D Tiles Next is an open specification, and contributions and feedback are encouraged. Over the next several months, our focus on 3D Tiles Next is to solidify the draft specifications and continue to facilitate ecosystem growth, keeping specification design pragmatic based on implementation experience.

Author: Done ✅

3D Tiles Next is an open specification, and contributions and feedback are encouraged. Over the next several months, our focus on 3D Tiles Next is to solidify the draft specifications and continue to facilitate ecosystem growth, keeping specification design pragmatic based on implementation experience.

> Issues and pull requests are welcome on this repository, in accordance with our guidelines for contributing, in [CONTRIBUTING.md](../CONTRIBUTING.md).

Reviewer: ```suggestion
Issues and pull requests are welcome on this repository. Please see our guidelines for contributing in [CONTRIBUTING.md](../CONTRIBUTING.md).
```
